frontend/src/views/Join.tsx

```diff
@@ -1,41 +1,54 @@
-import React, { useState, useEffect, ReactElement } from 'react';
+import React, { useState, ReactElement } from 'react';
+import { useLocation } from 'react-router-dom';
 import { Message } from 'stompjs';
-import ErrorMessage from '../components/core/Error';
-import { LargeText, Text, UserNicknameText } from '../components/core/Text';
-import { LargeCenterInputText, LargeInputButton } from '../components/core/Input';
+import { LargeText, UserNicknameText } from '../components/core/Text';
+import EnterNicknamePage from '../components/core/EnterNickname';
+import { errorHandler } from '../api/Error';
 import {
-  isValidNickname, connect, subscribe, deleteUser, User,
-  SOCKET_ENDPOINT, SUBSCRIBE_URL, addUser,
+  addUser, SUBSCRIBE_URL,
+  connect, deleteUser, SOCKET_ENDPOINT, subscribe, User,
 } from '../api/Socket';
+import ErrorMessage from '../components/core/Error';
+
+type JoinGamePageProps = {
+  initialPageState?: number,
+  initialNickname?: string;
+}
 
 function JoinGamePage() {
-  // Declare nickname state variable.
-  const [nickname, setNickname] = useState('');
+  // Grab initial state variables if navigated from the create page.
+  const location = useLocation<JoinGamePageProps>();
+  const joinGamePageProps: JoinGamePageProps = {};
+  if (location && location.state) {
+    joinGamePageProps.initialPageState = location.state.initialPageState;
+    joinGamePageProps.initialNickname = location.state.initialNickname;
+  }
 
   // Hold error text.
   const [error, setError] = useState('');
 
-  /**
-   * This is updated whenever the nickname changes.
-   */
-  const [validNickname, setValidNickname] = useState(false);
-  useEffect(() => {
-    setValidNickname(isValidNickname(nickname));
-  }, [nickname]);
-
-  // Variable to hold whether the user is focused on the text input field.
-  const [focusInput, setFocusInput] = useState(false);
-
   // Variable to hold the users on the page.
   const [users, setUsers] = useState<User[]>([]);
 
+  // Variable to hold whether the user is connected to the socket.
+  const [socketConnected, setSocketConnected] = useState(false);
+
   /**
    * Stores the current page state, where:
    * 0 = Enter room ID state (currently unused)
    * 1 = Enter nickname state
    * 2 = Waiting room state
    */
-  const [pageState, setPageState] = useState(1);
+  const [pageState, setPageState] = useState(joinGamePageProps.initialPageState || 1);
+
+  /**
+   * Nickname that is populated if the join page is on the waiting room stage.
+   * Set error if no nickname is passed in despite the waiting room stage.
+   */
+  const [nickname, setNickname] = useState(joinGamePageProps.initialNickname || '');
+  if (pageState === 2 && !nickname && !error) {
+    setError('No nickname was provided for the user in the waiting room.');
+  }
```

Comment:
I'm kind of confused, is it possible for a user to already be in the waiting room but not have a nickname?

Reply:
It shouldn't be possible -- I primarily included it to prevent issues from the developer's perspective. This could also be used to guard against malicious users, who somehow reach the second page state without a nickname.